docs/optimizing-javascript-code-loading.md

```diff
@@ -0,0 +1,222 @@
+// the SplashScreen component, and can unexpectedly break code elsewhere in
+// your app if you later decide to lazy-load SplashScreen.
+global.logger = new Logger();
+
+export function SplashScreen() {
+  // ...
+}
+```
+
+## Advanced: Call `require` inline
+
+Sometimes you may want to defer loading some code until you use it for the first time, without using `lazy` or an asynchronous `import()`. You can do this by using the [`require()`](https://metrobundler.dev/docs/module-api/#require) function where you would otherwise use a static `import` at the top of the file.
+
+```tsx title="VeryExpensive.tsx"
+import {Component} from 'react';
+import {Text} from 'react-native';
+// ... import some very expensive modules
+
+export default function VeryExpensive() {
+  // ... lots and lots of rendering logic
+  return <Text>Very Expensive Component</Text>;
+}
+```
+
+```tsx title="Optimized.tsx"
+import {useCallback, useState} from 'react';
+import {TouchableOpacity, View, Text} from 'react-native';
+// Usually we would write a static import:
+// import VeryExpensive from './VeryExpensive';
+
+let VeryExpensive = null;
+
+export default function Optimize() {
+  const [needsExpensive, setNeedsExpensive] = useState(false);
+  const didPress = useCallback(() => {
+    if (VeryExpensive == null) {
+      VeryExpensive = require('./VeryExpensive').default;
+    }
+
+    setNeedsExpensive(true);
+  }, []);
+
+  return (
+    <View style={{marginTop: 20}}>
+      <TouchableOpacity onPress={this.didPress}>
+        <Text>Load</Text>
+      </TouchableOpacity>
+      {this.state.needsExpensive ? <VeryExpensive /> : null}
+    </View>
+  );
+}
+```
+
+## Advanced: Automatically inline `require` calls
+
+If you use the React Native CLI to build your app, `require` calls (but not `import`s) will automatically be inlined for you, both in your code and inside any third-party packages (`node_modules`) you use.
+
+```tsx
+import {useCallback, useState} from 'react';
+import {TouchableOpacity, View, Text} from 'react-native';
+
+// This top-level require call will be evaluated lazily as part of the component below.
+const VeryExpensive = require('./VeryExpensive').default;
+
+export default function Optimize() {
+  const [needsExpensive, setNeedsExpensive] = useState(false);
+  const didPress = useCallback(() => {
+    setNeedsExpensive(true);
+  }, []);
+
+  return (
+    <View style={{marginTop: 20}}>
+      <TouchableOpacity onPress={this.didPress}>
+        <Text>Load</Text>
+      </TouchableOpacity>
+      {this.state.needsExpensive ? <VeryExpensive /> : null}
+    </View>
+  );
+}
+```
+
+:::info
+Some React Native frameworks disable this behavior. In particular, in Expo projects, `require` calls are not inlined by default. You can enable this optimization by editing your project's Metro config and setting `inlineRequires: true` in [`getTransformOptions`](https://metrobundler.dev/docs/configuration#gettransformoptions).
```

Comment:
Could we add a code block here with the full example? When I was first looking into this, it wasn't clear to me how to actually structure the JSON to add this option, something like:

```
// metro.config.js
{
  ...
  transformer: {
    getTransformOptions: {}
  }
}
```

... I actually can't even write the example without looking up a metro config file from somewhere